tools/docker/BUILD

```diff
@@ -15,10 +15,23 @@
 ################################################################################
 #
 load("@io_bazel_rules_docker//container:image.bzl", "container_image")
+load(
+    "@io_bazel_rules_docker//container:container.bzl",
+    "container_push",
+)
 
 container_image(
     name = "envoy",
     base = "@distroless_cc//image",
     tags = ["manual"],
     tars = ["//src/envoy:envoy_tar"],
 )
+
+container_push(
```

Comment:
We also need to build ubutu image along with distroless.

Reply:
What's wrong with distroless? We already statically link as much as we can.

Reply:
Nothing wrong with distroless, but we do publish ubuntu images.
I would like to publish images from here, so that istio/istio will only add statically linked binaries like "pilot-agent" and some other config.

We should add a test of running the binary from within the container, so as to not publish unrunnable images.

Reply:
If we want to build istio/istio image `FROM` this one it should be based on docker.io/istio/base to be consistent with other images

I am worried doing this may be pretty complicated though, in terms of making sure to use the right base image version, updating the dependency, etc. a lot of our tests rely on the `envoy` binary as well for example. What is the benefit of building the images here, and have we considered what istio/istio will look like if we do this?

Reply:
distroless is basically debian, so it's very close to ubuntu without all the garbage.
Agree about the tests, but that would require using docker to run images, which is something we want to avoid.

@howardjohn The goal is to avoid relying on istio to get working proxy images.

Reply:
I think having docker images built here is a good idea, just not 100% sure it's a good idea for istio/istio to consume them. Maybe it is, just needs some extra thought.

If we don't plan to directly build `FROM` these images them distroless seems fine to me?

Reply:
@mandarjog Added bionic-based build. This is very fast actually, so we can add as many as we want.

Reply:
@howardjohn Proxy should be the source of truth for all proxy build artifacts. Istio should stack any additions (pilot-agent, iptables) on top of it since Istio can produce statically linked go binaries with minimal system requirements. Gradually, as the agent becomes smaller, the proxy can become the complete source of truth for proxy images.

Reply:
I think we do need to build proxyv2 image `FROM` this. It's proxy repos responsibility to ensure that the container and the built image are compatible with each other.

proxyv2 build can add pilot-agent and whatever else on top of this.